toolkit/tools/imager/imager.go

```diff
@@ -66,13 +70,17 @@ func main() {
 	config, err := configuration.LoadWithAbsolutePaths(*configFile, *baseDirPath)
 	logger.PanicOnError(err, "Failed to load configuration file (%s) with base directory (%s)", *configFile, *baseDirPath)
 
+	// Run Preinstallation script
+	err = installutils.RunPreInstallScripts(config)
+	logger.PanicOnError(err, "Failed to preinstall scripts")
+
+	// Parse the kickstart partition file
+	err = configuration.ParseKickStartPartitionScheme(&config, kickstartPartitionFile)
```

Comment:
The config should be a return value rather than passed by the pointer.

Reply:
I think we generally pass by reference when we are editing anything in the config.